### PR TITLE
[BEAM-3135] Adding futures dependency to python SDK

### PR DESCRIPTION
**Pull request by angoenka**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
[sdk_worker](https://github.com/apache/beam/blob/a33c717dbb9b25485218b135a78b5cdb3f8239c9/sdks/python/apache_beam/runners/worker/sdk_worker.py#L30) is importing [futures](https://pypi.python.org/pypi/futures) module without declaring it in[ setup.py](https://github.com/apache/beam/blob/a33c717dbb9b25485218b135a78b5cdb3f8239c9/sdks/python/setup.py#L97)
Adding the direct dependency to futures to fix the dependencies of python sdk.

### CODE REVIEW

**Comment by angoenka** [comment]
Hi @aaltay, can you please take a look?

Here is the old dependency tree
apache-beam==2.1.1
  - avro [required: <2.0.0,>=1.8.1, installed: 1.8.2]
  - crcmod [required: <2.0,>=1.7, installed: 1.7]
  - dill [required: ==0.2.6, installed: 0.2.6]
  - grpcio [required: >=1.0,<2.0, installed: 1.7.0]
    - enum34 [required: >=1.0.4, installed: 1.1.6]
    - futures [required: >=2.2.0, installed: 3.1.1]
    - protobuf [required: >=3.3.0, installed: 3.3.0]
      - setuptools [required: Any, installed: 36.6.0]
      - six [required: >=1.9, installed: 1.10.0]
    - six [required: >=1.5.2, installed: 1.10.0]
  - httplib2 [required: >=0.8,<0.10, installed: 0.9.2]
  - mock [required: <3.0.0,>=1.0.1, installed: 2.0.0]
    - funcsigs [required: >=1, installed: 1.0.2]
    - pbr [required: >=0.11, installed: 3.1.1]
    - six [required: >=1.9, installed: 1.10.0]
  - oauth2client [required: >=2.0.1,<4.0.0, installed: 3.0.0]
    - httplib2 [required: >=0.9.1, installed: 0.9.2]
    - pyasn1 [required: >=0.1.7, installed: 0.3.7]
    - pyasn1-modules [required: >=0.0.5, installed: 0.1.5]
      - pyasn1 [required: >=0.3.4,<0.4.0, installed: 0.3.7]
    - rsa [required: >=3.1.4, installed: 3.4.2]
      - pyasn1 [required: >=0.1.3, installed: 0.3.7]
    - six [required: >=1.6.1, installed: 1.10.0]
  - protobuf [required: >=3.2.0,<=3.3.0, installed: 3.3.0]
    - setuptools [required: Any, installed: 36.6.0]
    - six [required: >=1.9, installed: 1.10.0]
  - pyyaml [required: <4.0.0,>=3.12, installed: 3.12]
  - six [required: <1.11,>=1.9, installed: 1.10.0]


and new dependency tree

apache-beam==2.3.0.dev0
  - avro [required: <2.0.0,>=1.8.1, installed: 1.8.2]
  - crcmod [required: <2.0,>=1.7, installed: 1.7]
  - dill [required: ==0.2.6, installed: 0.2.6]
  - **futures [required: >=2.2.0, installed: 3.1.1]**
  - grpcio [required: >=1.0,<2.0, installed: 1.7.0]
    - enum34 [required: >=1.0.4, installed: 1.1.6]
    - futures [required: >=2.2.0, installed: 3.1.1]
    - protobuf [required: >=3.3.0, installed: 3.3.0]
      - setuptools [required: Any, installed: 36.6.0]
      - six [required: >=1.9, installed: 1.10.0]
    - six [required: >=1.5.2, installed: 1.10.0]
  - httplib2 [required: >=0.8,<0.10, installed: 0.9.2]
  - mock [required: <3.0.0,>=1.0.1, installed: 2.0.0]
    - funcsigs [required: >=1, installed: 1.0.2]
    - pbr [required: >=0.11, installed: 3.1.1]
    - six [required: >=1.9, installed: 1.10.0]
  - oauth2client [required: >=2.0.1,<4.0.0, installed: 3.0.0]
    - httplib2 [required: >=0.9.1, installed: 0.9.2]
    - pyasn1 [required: >=0.1.7, installed: 0.3.7]
    - pyasn1-modules [required: >=0.0.5, installed: 0.1.5]
      - pyasn1 [required: >=0.3.4,<0.4.0, installed: 0.3.7]
    - rsa [required: >=3.1.4, installed: 3.4.2]
      - pyasn1 [required: >=0.1.3, installed: 0.3.7]
    - six [required: >=1.6.1, installed: 1.10.0]
  - protobuf [required: >=3.2.0,<=3.3.0, installed: 3.3.0]
    - setuptools [required: Any, installed: 36.6.0]
    - six [required: >=1.9, installed: 1.10.0]
  - pyyaml [required: <4.0.0,>=3.12, installed: 3.12]
  - six [required: <1.11,>=1.9, installed: 1.10.0]
  - typing [required: <3.7.0,>=3.6.0, installed: 3.6.2]

File: sdks/python/setup.py [hunk 1]
```diff
@@ -108,6 +108,7 @@ def get_version():
     # TODO(BEAM-2964): Remove the upper bound.
     'six>=1.9,<1.11',
     'typing>=3.6.0,<3.7.0',
+    'futures>=2.2.0',
```
Comment (aaltay) [inline sdks/python/setup.py]: Why are we not using the latest version?
Could you include an upper bound also (e.g `,<4`)

Comment (angoenka) [reply sdks/python/setup.py]: I imposed the min version requirement to be same as grpcio which was "futures [required: >=2.2.0, installed: 3.1.1]".
But having latest version makes sense.
Sure I can put the upperbound to <4

**Comment by aaltay** [comment]
LGTM, waiting for tests.

**Comment by aaltay** [comment]
retest this please

**Pull request closed without merging**